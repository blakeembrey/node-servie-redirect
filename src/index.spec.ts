import { compose } from 'throwback'
import { redirect } from './index'
import { Request, Response } from 'servie'
import { finalhandler } from 'servie-finalhandler'

describe('servie-redirect', () => {
  it('should send redirects', () => {
    const app = compose<Request, Response>([
      (req) => redirect(req, '/test')
    ])

    const req = new Request({ url: '/' })

    return app(req, finalhandler(req)).then((res) => {
      expect(res.toJSON()).toMatchSnapshot()
    })
  })
})
