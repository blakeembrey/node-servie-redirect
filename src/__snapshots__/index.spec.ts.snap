// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`servie-redirect should send redirects 1`] = `
Object {
  "body": undefined,
  "bytesTransferred": 0,
  "finished": false,
  "headers": Object {
    "location": "/test",
  },
  "started": false,
  "status": 302,
  "statusText": undefined,
  "trailers": Object {},
}
`;
